src/connector/src/sink/sqlserver.rs

```diff
@@ -0,0 +1,625 @@
+        schema: Schema,
+        pk_indices: Vec<usize>,
+        is_append_only: bool,
+    ) -> Result<Self> {
+        let sql_client = SqlClient::new(&config).await?;
+        let writer = Self {
+            config,
+            schema,
+            pk_indices,
+            is_append_only,
+            sql_client,
+            ops: vec![],
+        };
+        Ok(writer)
+    }
+
+    fn delete_one(&mut self, row: RowRef<'_>) {
+        // TODO: avoid into_owned_row
+        self.ops.push(SqlOp::Delete(row.into_owned_row()));
+    }
+
+    fn upsert_one(&mut self, row: RowRef<'_>) {
+        // TODO: avoid into_owned_row
+        if self.is_append_only {
+            self.ops.push(SqlOp::Insert(row.into_owned_row()));
+        } else {
+            self.ops.push(SqlOp::Merge(row.into_owned_row()));
+        }
+    }
+
+    async fn flush(&mut self) -> Result<()> {
+        use std::fmt::Write;
+        if self.ops.is_empty() {
+            return Ok(());
+        }
+        let mut query_str = String::new();
+        let col_num = self.schema.fields.len();
+        let mut next_param_id = 1;
+        let non_pk_col_indices = (0..col_num)
+            .filter(|idx| !self.pk_indices.contains(idx))
+            .collect::<Vec<usize>>();
+        let all_col_names = self
+            .schema
+            .fields
+            .iter()
+            .map(|f| f.name.as_str())
+            .collect::<Vec<_>>()
+            .join(",");
+        let all_source_col_names = self
+            .schema
+            .fields
+            .iter()
+            .map(|f| format!("SOURCE.{}", f.name))
+            .collect::<Vec<_>>()
+            .join(",");
+        let pk_match = self
+            .pk_indices
+            .iter()
+            .map(|idx| {
+                format!(
+                    "SOURCE.{}=TARGET.{}",
+                    &self.schema[*idx].name, &self.schema[*idx].name
+                )
+            })
+            .collect::<Vec<_>>()
+            .join(" AND ");
+        let param_placeholders = |param_id: &mut usize| {
+            let params = (*param_id..(*param_id + col_num))
+                .map(|i| format!("@P{}", i))
+                .collect::<Vec<_>>()
+                .join(",");
+            *param_id += col_num;
+            params
+        };
+        let set_all_source_col = non_pk_col_indices
+            .iter()
+            .map(|idx| {
+                format!(
+                    "{}=SOURCE.{}",
+                    &self.schema[*idx].name, &self.schema[*idx].name
+                )
+            })
+            .collect::<Vec<_>>()
+            .join(",");
+        // TODO: avoid repeating the SQL
+        for op in &self.ops {
+            match op {
+                SqlOp::Insert(_) => {
+                    write!(
+                        &mut query_str,
+                        "INSERT INTO {} ({}) VALUES ({})",
+                        self.config.table,
+                        all_col_names,
```

Comment:
becare to **quote** column names to support Uppercase column name

Reply:
fixed